clang/lib/Sema/CheckedCAnalysesPrepass.cpp

```diff
@@ -80,20 +93,50 @@ class PrepassHelper : public RecursiveASTVisitor<PrepassHelper> {
 
       // We add VarWithBounds to the set of all variables in whose bounds
       // expressions V occurs.
-      if (VarWithBounds) {
-        auto It = Info.BoundsVars.find(V);
-        if (It != Info.BoundsVars.end())
+      if (!VarWithBounds)
+        return true;
+
+      if (InBoundsExprLower) {
```

Comment:
It might be useful to create a helper function for adding `VarWithBounds` to the set for `V` in a given map (something like `AddBoundsSiblingField` in #1047). The method could be templated so it could replace `AddBoundsSiblingField`.